### PR TITLE
TEST-4 enrich tests

### PR DESCRIPTION
**Pull request by mtabasco**
Before (Integration Tests):
❌ Used greaterThan, lessThan, greaterThanOrEqual (11 violations)
❌ Used "tolerance" pattern allowing 100 wei rounding error
❌ Didn't track block numbers for precise calculations
After (Integration Tests):
✅ All assertions use exact .to.equal() with calculated expected values
✅ Block numbers tracked precisely for fee calculations
✅ Every balance assertion references SPEC.md formulas

Added 3 New Edge Case Tests:
File: test/unit/SSVClusters/updateClusterBalance.test.ts
Test 1: Multi-Validator Liquidated Cluster (Lines 463-527)
Test 2: EB Decrease on Liquidated Cluster (Lines 529-602)
Test 3: Implicit→Explicit EB Transition (Lines 604-653)

### CODE REVIEW

File: test/common/helpers.ts [hunk 1]
```diff
@@ -195,7 +197,7 @@ export async function registerDefaultCluster(
   );
 
   return {
-    cluster, validatorKey, operatorIds
+    cluster, validatorKey, operatorIds, receiptRegister
```
Comment (mtabasco) [inline test/common/helpers.ts:200]: @yurii-ssv do you think adding tx receipt here is useful? I only use it to get the block number in one test...

Comment (yurii-ssv) [reply test/common/helpers.ts:200]: Yes, why not. Can be very helpful when we will start addressing #435

**Review by yurii-ssv — APPROVED** [review]
